{"nbformat":4,"nbformat_minor":5,"metadata":{"kernelspec":{"name":"python3","display_name":"Yandex DataSphere Kernel","language":"python"},"language_info":{"pygments_lexer":"ipython3","nbconvert_exporter":"python","name":"python","codemirror_mode":{"version":3,"name":"ipython"},"mimetype":"text/x-python","version":"3.7.7","file_extension":".py"},"notebookId":"27566d2c-1853-4137-87dd-860d62edf963"},"cells":[{"cell_type":"markdown","source":"# Устанавливаем всё необходимое","metadata":{"cellId":"w5wxnuv1pb8j0js5gogw"}},{"cell_type":"code","source":"#!c1.4\n%cd /home/jupyter/work/resources\n!git clone https://github.com/ivan7022/MakeUHappyChatBot\n%cd MakeUHappyChatBot","metadata":{"cellId":"g2y4v0xwmv95oirqrx6nn","trusted":true},"outputs":[{"output_type":"stream","name":"stdout","text":"fatal: destination path 'MakeUHappyChatBot' already exists and is not an empty directory.\n"},{"output_type":"error","ename":"Exception","evalue":"Process exited with code 128","traceback":["\u001B[0;31m---------------------------------------------------------------------------\u001B[0m","\u001B[0;31mException\u001B[0m                                 Traceback (most recent call last)","\u001B[0;32m<ipython-input-1-c9b3aabea686>\u001B[0m in \u001B[0;36m<module>\u001B[0;34m\u001B[0m\n\u001B[0;32m----> 1\u001B[0;31m \u001B[0mget_ipython\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0msystem\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m'git clone https://github.com/ivan7022/MakeUHappyChatBot'\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m\u001B[1;32m      2\u001B[0m \u001B[0mget_ipython\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mrun_line_magic\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m'cd'\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0;34m'MakeUHappyChatBot'\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m      3\u001B[0m \u001B[0;31m#\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n","\u001B[0;32m/kernel/lib/python3.7/site-packages/ml_kernel/kernel.py\u001B[0m in \u001B[0;36m<lambda>\u001B[0;34m(code)\u001B[0m\n\u001B[1;32m    255\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    256\u001B[0m         \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_script_executor\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0mScriptExecutor\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mProcessHandler\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0msystem\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m--> 257\u001B[0;31m         \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mshell\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0msystem\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0;32mlambda\u001B[0m \u001B[0mcode\u001B[0m\u001B[0;34m:\u001B[0m \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_script_executor\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mexecute\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m\"bash\"\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0mcode\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m\u001B[1;32m    258\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    259\u001B[0m     \u001B[0;32mdef\u001B[0m \u001B[0m_init_message_handlers\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mself\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m:\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n","\u001B[0;32m/kernel/lib/python3.7/site-packages/ml_kernel/script_executor.py\u001B[0m in \u001B[0;36mexecute\u001B[0;34m(self, lang, code)\u001B[0m\n\u001B[1;32m     33\u001B[0m         \u001B[0mreturn_code\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_system\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mcode\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m     34\u001B[0m         \u001B[0;32mif\u001B[0m \u001B[0mreturn_code\u001B[0m \u001B[0;34m!=\u001B[0m \u001B[0;36m0\u001B[0m\u001B[0;34m:\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m---> 35\u001B[0;31m             \u001B[0;32mraise\u001B[0m \u001B[0mException\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m'Process exited with code %d'\u001B[0m \u001B[0;34m%\u001B[0m \u001B[0mreturn_code\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m","\u001B[0;31mException\u001B[0m: Process exited with code 128"]}],"execution_count":8},{"cell_type":"code","source":"#!c1.4\n%pip install -r requirements.txt","metadata":{"cellId":"u0go9d7xxq1h8hbu66e0o","trusted":true},"outputs":[{"output_type":"stream","name":"stdout","text":"Defaulting to user installation because normal site-packages is not writeable\nRequirement already satisfied: emoji in /home/jupyter/.local/lib/python3.7/site-packages (from -r MakeUHappyChatBot/requirements.txt (line 1)) (1.2.0)\nRequirement already satisfied: google_trans_new in /home/jupyter/.local/lib/python3.7/site-packages (from -r MakeUHappyChatBot/requirements.txt (line 2)) (1.1.9)\nCollecting hnswlib\n  Downloading hnswlib-0.5.1.tar.gz (29 kB)\n    Preparing wheel metadata ... \u001B[?25lerror\n\u001B[31m    ERROR: Command errored out with exit status 1:\n     command: /usr/bin/python3.7 /usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp6kzqn5yp\n         cwd: /tmp/pip-install-j3ulbp5e/hnswlib_51c3d98a25db4689b9e21c5ce8b33aa2\n    Complete output (14 lines):\n    Traceback (most recent call last):\n      File \"/usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/_in_process.py\", line 280, in <module>\n        main()\n      File \"/usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/_in_process.py\", line 263, in main\n        json_out['return_val'] = hook(**hook_input['kwargs'])\n      File \"/usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/_in_process.py\", line 133, in prepare_metadata_for_build_wheel\n        return hook(metadata_directory, config_settings)\n      File \"/kernel/lib/python3.7/site-packages/setuptools/build_meta.py\", line 161, in prepare_metadata_for_build_wheel\n        self.run_setup()\n      File \"/kernel/lib/python3.7/site-packages/setuptools/build_meta.py\", line 145, in run_setup\n        exec(compile(code, __file__, 'exec'), locals())\n      File \"setup.py\", line 5, in <module>\n        import pybind11\n    ModuleNotFoundError: No module named 'pybind11'\n    ----------------------------------------\u001B[0m\n\u001B[33mWARNING: Discarding https://files.pythonhosted.org/packages/03/8c/3e0e608278b740f2a78ba76ba406dbecc8b7d3ce8cfb858580f13ea04930/hnswlib-0.5.1.tar.gz#sha256=5598c0a1a6eeb577bc271cac59983c89eb4c1beb1b7e75ba50068e2dc0c038f2 (from https://pypi.org/simple/hnswlib/). Command errored out with exit status 1: /usr/bin/python3.7 /usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmp6kzqn5yp Check the logs for full command output.\u001B[0m\n\u001B[?25h  Downloading hnswlib-0.5.0.tar.gz (29 kB)\n    Preparing wheel metadata ... \u001B[?25lerror\n\u001B[31m    ERROR: Command errored out with exit status 1:\n     command: /usr/bin/python3.7 /usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpp4ywvmfu\n         cwd: /tmp/pip-install-j3ulbp5e/hnswlib_e269993ed94c4880b7d19292dfa24257\n    Complete output (14 lines):\n    Traceback (most recent call last):\n      File \"/usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/_in_process.py\", line 280, in <module>\n        main()\n      File \"/usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/_in_process.py\", line 263, in main\n        json_out['return_val'] = hook(**hook_input['kwargs'])\n      File \"/usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/_in_process.py\", line 133, in prepare_metadata_for_build_wheel\n        return hook(metadata_directory, config_settings)\n      File \"/kernel/lib/python3.7/site-packages/setuptools/build_meta.py\", line 161, in prepare_metadata_for_build_wheel\n        self.run_setup()\n      File \"/kernel/lib/python3.7/site-packages/setuptools/build_meta.py\", line 145, in run_setup\n        exec(compile(code, __file__, 'exec'), locals())\n      File \"setup.py\", line 5, in <module>\n        import pybind11\n    ModuleNotFoundError: No module named 'pybind11'\n    ----------------------------------------\u001B[0m\n\u001B[33mWARNING: Discarding https://files.pythonhosted.org/packages/fb/ff/a02bb1971d6034ff268bb1554009ac074d9d58434129c50bf136e7f39897/hnswlib-0.5.0.tar.gz#sha256=5a3a3d79442852b0587aec0c773a16a20e3a3b105efa229301878e78012b9901 (from https://pypi.org/simple/hnswlib/). Command errored out with exit status 1: /usr/bin/python3.7 /usr/local/lib/python3.7/dist-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpp4ywvmfu Check the logs for full command output.\u001B[0m\n\u001B[?25h  Downloading hnswlib-0.4.0.tar.gz (17 kB)\nRequirement already satisfied: numpy in /kernel/lib/python3.7/site-packages (from -r MakeUHappyChatBot/requirements.txt (line 4)) (1.19.4)\nCollecting pymystem3\n  Downloading pymystem3-0.2.0-py3-none-any.whl (10 kB)\nRequirement already satisfied: requests in /kernel/lib/python3.7/site-packages (from -r MakeUHappyChatBot/requirements.txt (line 6)) (2.25.1)\nRequirement already satisfied: torch in /usr/local/lib/python3.7/dist-packages (from -r MakeUHappyChatBot/requirements.txt (line 7)) (1.6.0)\nRequirement already satisfied: tqdm in /home/jupyter/.local/lib/python3.7/site-packages (from -r MakeUHappyChatBot/requirements.txt (line 8)) (4.49.0)\nRequirement already satisfied: transformers in /home/jupyter/.local/lib/python3.7/site-packages (from -r MakeUHappyChatBot/requirements.txt (line 9)) (4.6.1)\nCollecting pybind11>=2.0\n  Downloading pybind11-2.6.2-py2.py3-none-any.whl (191 kB)\n\u001B[K     |████████████████████████████████| 191 kB 4.3 MB/s \n\u001B[?25hRequirement already satisfied: chardet<5,>=3.0.2 in /home/jupyter/.local/lib/python3.7/site-packages (from requests->-r MakeUHappyChatBot/requirements.txt (line 6)) (3.0.4)\nRequirement already satisfied: urllib3<1.27,>=1.21.1 in /kernel/lib/python3.7/site-packages (from requests->-r MakeUHappyChatBot/requirements.txt (line 6)) (1.26.5)\nRequirement already satisfied: certifi>=2017.4.17 in /kernel/lib/python3.7/site-packages (from requests->-r MakeUHappyChatBot/requirements.txt (line 6)) (2020.12.5)\nRequirement already satisfied: idna<3,>=2.5 in /kernel/lib/python3.7/site-packages (from requests->-r MakeUHappyChatBot/requirements.txt (line 6)) (2.10)\nRequirement already satisfied: future in /usr/local/lib/python3.7/dist-packages (from torch->-r MakeUHappyChatBot/requirements.txt (line 7)) (0.18.2)\nRequirement already satisfied: tokenizers<0.11,>=0.10.1 in /home/jupyter/.local/lib/python3.7/site-packages (from transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (0.10.3)\nRequirement already satisfied: filelock in /usr/local/lib/python3.7/dist-packages (from transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (3.0.12)\nRequirement already satisfied: sacremoses in /usr/local/lib/python3.7/dist-packages (from transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (0.0.43)\nRequirement already satisfied: importlib-metadata in /kernel/lib/python3.7/site-packages (from transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (4.1.0)\nRequirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.7/dist-packages (from transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (2021.3.17)\nRequirement already satisfied: huggingface-hub==0.0.8 in /home/jupyter/.local/lib/python3.7/site-packages (from transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (0.0.8)\nRequirement already satisfied: packaging in /kernel/lib/python3.7/site-packages (from transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (20.9)\nRequirement already satisfied: typing-extensions>=3.6.4 in /kernel/lib/python3.7/site-packages (from importlib-metadata->transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (3.10.0.0)\nRequirement already satisfied: zipp>=0.5 in /kernel/lib/python3.7/site-packages (from importlib-metadata->transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (3.4.1)\nRequirement already satisfied: pyparsing>=2.0.2 in /kernel/lib/python3.7/site-packages (from packaging->transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (2.4.7)\nRequirement already satisfied: six in /kernel/lib/python3.7/site-packages (from sacremoses->transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (1.16.0)\nRequirement already satisfied: click in /usr/local/lib/python3.7/dist-packages (from sacremoses->transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (7.1.2)\nRequirement already satisfied: joblib in /usr/local/lib/python3.7/dist-packages (from sacremoses->transformers->-r MakeUHappyChatBot/requirements.txt (line 9)) (1.0.1)\nBuilding wheels for collected packages: hnswlib\n  Building wheel for hnswlib (setup.py) ... \u001B[?25lerror\n\u001B[31m  ERROR: Command errored out with exit status 1:\n   command: /usr/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '\"'\"'/tmp/pip-install-j3ulbp5e/hnswlib_5e3e5f8d6e0744efb3b23310614dbdb8/setup.py'\"'\"'; __file__='\"'\"'/tmp/pip-install-j3ulbp5e/hnswlib_5e3e5f8d6e0744efb3b23310614dbdb8/setup.py'\"'\"';f=getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__);code=f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' bdist_wheel -d /tmp/pip-wheel-jmwalkhk\n       cwd: /tmp/pip-install-j3ulbp5e/hnswlib_5e3e5f8d6e0744efb3b23310614dbdb8/\n  Complete output (39 lines):\n  running bdist_wheel\n  running build\n  running build_ext\n  creating tmp\n  x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fdebug-prefix-map=/build/python3.7-a56wZI/python3.7-3.7.10=. -fstack-protector-strong -Wformat -Werror=format-security -g -fdebug-prefix-map=/build/python3.7-a56wZI/python3.7-3.7.10=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.7m -c /tmp/tmp_nos6gq7.cpp -o tmp/tmp_nos6gq7.o -std=c++14\n  x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fdebug-prefix-map=/build/python3.7-a56wZI/python3.7-3.7.10=. -fstack-protector-strong -Wformat -Werror=format-security -g -fdebug-prefix-map=/build/python3.7-a56wZI/python3.7-3.7.10=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.7m -c /tmp/tmppy06jm1l.cpp -o tmp/tmppy06jm1l.o -fvisibility=hidden\n  Traceback (most recent call last):\n    File \"<string>\", line 1, in <module>\n    File \"/tmp/pip-install-j3ulbp5e/hnswlib_5e3e5f8d6e0744efb3b23310614dbdb8/setup.py\", line 116, in <module>\n      zip_safe=False,\n    File \"/kernel/lib/python3.7/site-packages/setuptools/__init__.py\", line 153, in setup\n      return distutils.core.setup(**attrs)\n    File \"/usr/lib/python3.7/distutils/core.py\", line 148, in setup\n      dist.run_commands()\n    File \"/usr/lib/python3.7/distutils/dist.py\", line 966, in run_commands\n      self.run_command(cmd)\n    File \"/usr/lib/python3.7/distutils/dist.py\", line 985, in run_command\n      cmd_obj.run()\n    File \"/usr/local/lib/python3.7/dist-packages/wheel/bdist_wheel.py\", line 299, in run\n      self.run_command('build')\n    File \"/usr/lib/python3.7/distutils/cmd.py\", line 313, in run_command\n      self.distribution.run_command(command)\n    File \"/usr/lib/python3.7/distutils/dist.py\", line 985, in run_command\n      cmd_obj.run()\n    File \"/usr/lib/python3.7/distutils/command/build.py\", line 135, in run\n      self.run_command(cmd_name)\n    File \"/usr/lib/python3.7/distutils/cmd.py\", line 313, in run_command\n      self.distribution.run_command(command)\n    File \"/usr/lib/python3.7/distutils/dist.py\", line 985, in run_command\n      cmd_obj.run()\n    File \"/kernel/lib/python3.7/site-packages/setuptools/command/build_ext.py\", line 79, in run\n      _build_ext.run(self)\n    File \"/kernel/lib/python3.7/site-packages/Cython/Distutils/old_build_ext.py\", line 186, in run\n      _build_ext.build_ext.run(self)\n    File \"/usr/lib/python3.7/distutils/command/build_ext.py\", line 340, in run\n      self.build_extensions()\n    File \"/tmp/pip-install-j3ulbp5e/hnswlib_5e3e5f8d6e0744efb3b23310614dbdb8/setup.py\", line 88, in build_extensions\n      import pybind11\n  ModuleNotFoundError: No module named 'pybind11'\n  ----------------------------------------\u001B[0m\n\u001B[31m  ERROR: Failed building wheel for hnswlib\u001B[0m\n\u001B[?25h  Running setup.py clean for hnswlib\nFailed to build hnswlib\nInstalling collected packages: pybind11, pymystem3, hnswlib\n\u001B[33m  WARNING: The script pybind11-config is installed in '/home/jupyter/.local/bin' which is not on PATH.\n  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.\u001B[0m\n    Running setup.py install for hnswlib ... \u001B[?25ldone\n\u001B[33m  DEPRECATION: hnswlib was installed using the legacy 'setup.py install' method, because a wheel could not be built for it. A possible replacement is to fix the wheel build issue reported above. You can find discussion regarding this at https://github.com/pypa/pip/issues/8368.\u001B[0m\n\u001B[?25hSuccessfully installed hnswlib-0.4.0 pybind11-2.6.2 pymystem3-0.2.0\n\u001B[33mWARNING: You are using pip version 21.0.1; however, version 21.1.2 is available.\nYou should consider upgrading via the '/usr/bin/python3.7 -m pip install --upgrade pip' command.\u001B[0m\n"}],"execution_count":2},{"cell_type":"code","source":"%cd prod","metadata":{"cellId":"63owxngmvxgg3usewlvdrq"},"outputs":[],"execution_count":null},{"cell_type":"code","source":"#!g1.1\n!python3 __init__.py","metadata":{"cellId":"yvmy922hctj1ny2bl0t8fe","trusted":true},"outputs":[{"output_type":"stream","name":"stdout","text":"Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.\nSpecial tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.\nChat-bot started\nReceived message: как дела?\nLength of chat history id 1\nReceived message: как дела?\nLength of chat history id 1\nReceived message: у меня всё классно, а у тебя как?\nLength of chat history id 1\nReceived message: круто!\nLength of chat history id 1\nReceived message: погода просто супер!\nLength of chat history id 1\nReceived message: а где ты живёшь?\nLength of chat history id 1\n^C\nChat-bot stopped\n"},{"output_type":"error","ename":"Exception","evalue":"Process exited with code -1","traceback":["\u001B[0;31m---------------------------------------------------------------------------\u001B[0m","\u001B[0;31mException\u001B[0m                                 Traceback (most recent call last)","\u001B[0;32m<ipython-input-7-61f4007550bf>\u001B[0m in \u001B[0;36m<module>\u001B[0;34m\u001B[0m\n\u001B[0;32m----> 1\u001B[0;31m \u001B[0mget_ipython\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0msystem\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m'python3 __init__.py'\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m\u001B[1;32m      2\u001B[0m \u001B[0;31m#\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n","\u001B[0;32m/kernel/lib/python3.7/site-packages/ml_kernel/kernel.py\u001B[0m in \u001B[0;36m<lambda>\u001B[0;34m(code)\u001B[0m\n\u001B[1;32m    255\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    256\u001B[0m         \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_script_executor\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0mScriptExecutor\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mProcessHandler\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0msystem\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m--> 257\u001B[0;31m         \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mshell\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0msystem\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0;32mlambda\u001B[0m \u001B[0mcode\u001B[0m\u001B[0;34m:\u001B[0m \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_script_executor\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0mexecute\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m\"bash\"\u001B[0m\u001B[0;34m,\u001B[0m \u001B[0mcode\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m\u001B[1;32m    258\u001B[0m \u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m    259\u001B[0m     \u001B[0;32mdef\u001B[0m \u001B[0m_init_message_handlers\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mself\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m:\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n","\u001B[0;32m/kernel/lib/python3.7/site-packages/ml_kernel/script_executor.py\u001B[0m in \u001B[0;36mexecute\u001B[0;34m(self, lang, code)\u001B[0m\n\u001B[1;32m     33\u001B[0m         \u001B[0mreturn_code\u001B[0m \u001B[0;34m=\u001B[0m \u001B[0mself\u001B[0m\u001B[0;34m.\u001B[0m\u001B[0m_system\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0mcode\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[1;32m     34\u001B[0m         \u001B[0;32mif\u001B[0m \u001B[0mreturn_code\u001B[0m \u001B[0;34m!=\u001B[0m \u001B[0;36m0\u001B[0m\u001B[0;34m:\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0;32m---> 35\u001B[0;31m             \u001B[0;32mraise\u001B[0m \u001B[0mException\u001B[0m\u001B[0;34m(\u001B[0m\u001B[0;34m'Process exited with code %d'\u001B[0m \u001B[0;34m%\u001B[0m \u001B[0mreturn_code\u001B[0m\u001B[0;34m)\u001B[0m\u001B[0;34m\u001B[0m\u001B[0;34m\u001B[0m\u001B[0m\n\u001B[0m","\u001B[0;31mException\u001B[0m: Process exited with code -1"]}],"execution_count":22}]}